### PR TITLE
Add obj.makePhot method

### PR DESCRIPTION
**Pull request by rmjarvis**
This PR adds a way to draw photons without actually accumulating them onto an image.

It is functionally equivalent to `_, photons = obj.drawPhot(im, ...)` but without requiring the image.

@cwwalter

### CODE REVIEW

**Comment by rmjarvis** [comment]
>  Just one question: would it be possible for drawPhot to use makePhot? Too tricky with the resume I'm guessing?

That was originally my plan, but with all the maxN and resume stuff, if seemed a little too much to untangle cleanly.  This seemed simpler, and the unit tests make sure they are equivalent for their overlapping use cases.

File: galsim/gsobject.py [hunk 1]
```diff
@@ -2067,6 +2067,101 @@ def _calculate_nphotons(self, n_photons, poisson_flux, max_extra_noise, rng):
         return iN, g
 
 
+    def makePhot(self, n_photons=0, rng=None, max_extra_noise=0., poisson_flux=None,
+                 sensor=None, surface_ops=(), maxN=None, orig_center=PositionI(0,0),
```
Comment (cwwalter) [inline galsim/gsobject.py]: Sensor isn't in the parameter list.

File: galsim/gsobject.py [hunk 2]
```diff
@@ -2067,6 +2067,101 @@ def _calculate_nphotons(self, n_photons, poisson_flux, max_extra_noise, rng):
...
+                            applied in order before accumulating the photons on the sensor.
+                            [default: ()]
+            orig_center:    The position of the image center in the original image coordinates.
+                            [default: (0,0)]
+            local_wcs:      The local wcs in the original image. [default: None]
+
+        Returns:
+            - a `PhotonArray` with the data about the photons.
+        """
+        from .sensor import Sensor
```
Comment (cwwalter) [inline galsim/gsobject.py:2128]: Is this necessary in this function?

File: galsim/gsobject.py (hunk 1 above)
Comment (rmjarvis) [reply galsim/gsobject.py]: Thanks.  Also maxN.  Fixed.

File: galsim/gsobject.py (hunk 2 above)
Comment (rmjarvis) [reply galsim/gsobject.py:2128]: Thanks.  No not needed here.

**Comment by rmjarvis** [comment]
> Also, just for my education/curiosity: Why all the changes to " rng=rng.duplicate()" in the tests? Was it not giving a reproducible sequence before?

These tests check that drawPhot and makePhot produce the same exact photons.  So the first time, for drawPhot, I call rng.duplicate() to fork the RNG sequence.  Then when I do it again with makePhot, the rng runs through the exact same sequence again, so should produce the same photons.

**Pull request merged**